pkg/apis/cr/v1alpha1/types.go

```diff
@@ -196,27 +212,42 @@ var _ runtime.Object = (*Blueprint)(nil)
 type Blueprint struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata"`
-	Actions           map[string]*BlueprintAction `json:"actions,omitempty"`
+	// Actions is the list of actions constructing the Blueprint.
+	Actions map[string]*BlueprintAction `json:"actions,omitempty"`
 }
 
 // BlueprintAction describes the set of phases that constitute an action.
 type BlueprintAction struct {
-	Name               string              `json:"name"`
-	Kind               string              `json:"kind"`
-	ConfigMapNames     []string            `json:"configMapNames,omitempty"`
-	SecretNames        []string            `json:"secretNames,omitempty"`
-	InputArtifactNames []string            `json:"inputArtifactNames,omitempty"`
-	OutputArtifacts    map[string]Artifact `json:"outputArtifacts,omitempty"`
-	Phases             []BlueprintPhase    `json:"phases,omitempty"`
-	DeferPhase         *BlueprintPhase     `json:"deferPhase,omitempty"`
+	// Name contains the name of the action.
+	Name string `json:"name"`
+	// Kind contains the resource on which this action has to be performed. Kind field is now deprecated.
```

Comment:
Let's remove second sentence from here. We still use this in our blueprints. And if we decide that we want to remove that from blueprint we can update this.

Reply:
Made the changes.